pyoverkiz/models.py

```diff
@@ -181,7 +189,17 @@ def __init__(
         self.data_properties = data_properties
         self.type = ProductType(type)
         self.place_oid = place_oid
-        self.protocol = Protocol(self.device_url.split(":")[0])
+
+        match = re.search(DEVICE_REGEX, device_url)
+
+        self.protocol = Protocol(match.group('protocol'))
+        self.gateway_id = match.group('gatewayId')
+        self.device_address = match.group('deviceAddress')
+
+        if match.group('subsystemId'):
+            self.subsystem_id = int(match.group('subsystemId'))
+            if self.subsystem_id > 1:
+                self.is_sub_device = True
```

Comment:
What's the goal of such property?

Reply:
There: https://github.com/home-assistant/core/blob/dev/homeassistant/components/overkiz/entity.py#L37